grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcHealthCheckServiceTest.java

```diff
@@ -0,0 +1,343 @@
+                                    .build();
+        response = client.check(request);
+        assertThat(response.getStatus()).isEqualTo(ServingStatus.SERVING);
+
+        final StatusRuntimeException exception = assertThrows(StatusRuntimeException.class, () -> client.check(
+                HealthCheckRequest.newBuilder()
+                                  .setService("com.linecorp.armeria.grpc.testing.NotFoundTestService")
+                                  .build()));
+        assertThat(exception.getStatus().getCode()).isEqualTo(Status.NOT_FOUND.getCode());
+        assertThat(exception.getMessage()).isEqualTo(
+                "NOT_FOUND: The service name(=com.linecorp.armeria.grpc.testing.NotFoundTestService) is not registered in this service");
+    }
+
+    @Test
+    void watch() throws Exception {
+        final HealthCheckRequest request = HealthCheckRequest.getDefaultInstance();
+        final HealthStub client = Clients.newClient(
+                server.httpUri(GrpcSerializationFormats.PROTO),
+                HealthStub.class);
+        final StreamRecorder<HealthCheckResponse> recorder = StreamRecorder.create();
+        client.watch(request, recorder);
+        TimeUnit.SECONDS.sleep(1);
+        serveHealth.setHealthy(false);
+        TimeUnit.SECONDS.sleep(1);
+        serveHealth.setHealthy(true);
+        TimeUnit.SECONDS.sleep(1);
+        recorder.onCompleted();
+        final List<ServingStatus> responses = recorder.getValues()
+                                                      .stream()
+                                                      .map(HealthCheckResponse::getStatus)
+                                                      .collect(Collectors.toList());
+        assertThat(responses).containsExactlyElementsOf(
+                ImmutableList.of(ServingStatus.SERVING, ServingStatus.NOT_SERVING, ServingStatus.SERVING));
+    }
+
+    @ParameterizedTest
+    @MethodSource("checkServingStatusArguments")
+    void checkServingStatus(GrpcHealthCheckService grpcHealthCheckService,
+                            String serviceName,
+                            boolean serverIsHealthy,
+                            ServingStatus expected) throws Exception {
+        grpcHealthCheckService.changeServerStatus(serverIsHealthy);
+        assertThat(grpcHealthCheckService.checkServingStatus(serviceName)).isEqualTo(expected);
+    }
+
+    private static Stream<Arguments> checkServingStatusArguments() {
+        return Stream.of(
+                // request: empty server name
+                // condition: server is healthy, health checker is not given.
+                // response: SERVING
+                Arguments.of(new GrpcHealthCheckService(
+                                     Collections.emptySet(),
+                                     Collections.emptyMap(),
+                                     Collections.emptyList()),
```

Comment:
nit: We prefer to use Guava's `ImmutableList.of()`, `ImmutableSet.of()` and `ImmutableMap.of()`.